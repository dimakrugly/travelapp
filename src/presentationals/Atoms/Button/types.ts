import { TextStyle, ViewStyle } from 'react-native';
export interface ButtonProps {
  title: string,
  onPress: ()=>void,
}

export interface ButtonStyles {
  container: ViewStyle,
  button: ViewStyle,
  title: TextStyle,
}
