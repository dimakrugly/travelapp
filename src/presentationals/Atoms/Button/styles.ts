import { StyleSheet } from 'react-native';
import { ButtonStyles } from './types';



export const styles  = StyleSheet.create<ButtonStyles>({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    borderRadius: 15,
    height: 48,
    width: '100%',
    backgroundColor: 'black',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: 'white',
  },
});
