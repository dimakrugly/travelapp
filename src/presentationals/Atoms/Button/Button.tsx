import { TouchableOpacity, View, Text } from 'react-native';
import React from 'react';
import { ButtonProps } from './types';
import { styles } from './styles';


export const AppButton: React.FC<ButtonProps> = ({ title, onPress }) => {
  return (
    <View style={styles.container} >
      <TouchableOpacity style={styles.button}  onPress={onPress} >
        <Text style={styles.title}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};
