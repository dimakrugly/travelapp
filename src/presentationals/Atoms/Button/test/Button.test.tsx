import {render} from '@testing-library/react-native';
import {AppButton} from '../Button';

describe('AppButton', () => {
  it('Match snapshot', () => {
    const tree = render(<AppButton title={'title'} onPress={() => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
