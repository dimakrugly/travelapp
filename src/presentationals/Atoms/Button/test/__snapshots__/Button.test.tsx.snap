// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton Match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderRadius": 15,
        "height": 48,
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
        }
      }
    >
      title
    </Text>
  </View>
</View>
`;
